pants_jupyter_plugin/pex.py

```diff
@@ -67,23 +94,60 @@ def unmount(self) -> Iterator[Path]:
                         del sys.modules[name]
             yield pex_sys_path_entry
 
-    def mount_pex(self, pex_to_mount: Path) -> Iterator[Path]:
+    def mount(self, pex_to_mount: Path) -> Iterator[Path]:
         """Mounts the contents of the given PEX on the sys.path for importing."""
-        venv = (
-            self._CACHE
-            / "venvs"
-            / hashlib.sha1(pex_to_mount.read_bytes()).hexdigest()
-            / pex_to_mount.name
-        )
+        current_interpreter = Path(sys.executable)
+
+        hasher = hashlib.sha1()
+        hasher.update(current_interpreter.read_bytes())
+        hasher.update(pex_to_mount.read_bytes())
+        fingerprint = hasher.hexdigest()
+
+        venv = self._CACHE / "venvs" / fingerprint / pex_to_mount.name
         with creation_lock(venv) as locked:
             if locked:
```

Comment:
This pattern was here before, but it seems strange to fall through if we fail to acquire the lock. If we lost the race, might we have a partially constructed venv?

Reply:
Its a blocking lock. You only fall through after the last line of this block is executed by the lock holder. That line is `run_pex_tool(args=["venv", str(venv)])` and Pex `venv` is ... not atomic itself - unlike ~every dir creation operation in Pex. Because of that last bit - good catch. Although I will fix that over in Pex, I won't wait for that fix.

Reply:
Fixed.